### PR TITLE
Core build/storybook fixes

### LINKED ISSUE #4510: [CORE DOX] Update Beta definition on website/storybook
## Summary

Our current messaging around clarity core been" ETA" is confusing internal teams that are unclear whether the web components are even available through NPM.

Based on a discussion today, we want to update our messaging in such a way that we have the flexibility to make breaking changes to works-in-progress but also clearly delineate components and utilities whose API is stable.

### Change 1

`https://clarity.design/storybook/core/?path=/story/welcome-clarity--core`

• Remove the beta icon from the welcome page

• Change the second paragraph to read:
```
Some of our components and utilities are currently offered as previews. This means that these components and utilities are in active development but ready to try out! We are hoping you will share your feedback on these new components. You can read more about Core and its goals in our Medium blog post.
```
(all links remain the same)

### Change 2

The "experimental" header should be changed to say "Preview". This should also impact the URLs in storybook.

### Change 3

Update the alert in all of the experimental pages to read:

```
This component or utility is offered as a preview. This means we are currently working on it and seeking feedback. Please be aware that this component or utility may have breaking changes before we finish working on it.
```

### PR DESCRIPTION
**Pull request by coryrylan**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [x] CI related changes
- [x] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the new behavior?
- Update layouts to support additional slot/shadow dom use cases
- Update layouts to support nesting of horizontal/vertical layouts without containers
- Update layouts wrap API for additional options coming in forms
- Update storybook version
- Fix API generation issue with Core and package name/version generation
- Add support to hide experimental/internal APIs in storybook during deployment
- Update storybook getting started docs and remove beta references closes #4510 
- Fix minor issues with styles from base component style with `contain: layout`
- Fix button link touch target bug
- Add `spreadProps` directive and storybook args utils for easy API demos (buttons only updated, forms will be in next pr)
- Fix native style focus to be more consistent with new Chromium focus styles
- Fix storybook modal demo style leak

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -13,19 +12,20 @@
...
+    "build:package": "cpy '**/package.json' '../dist/core/' --cwd=./src --parents && cpy ./package.json ./dist/core/ && cpy ./README.md ./dist/core/; replace '0.0.0' $npm_package_version ./dist/core/package.json; replace '@VERSION' $npm_package_version ./dist/core/internal/utils/global.js && node ./package.js",
     "build:ts": "tsc --b ./tsconfig.project.json --force",
-    "build:ts:watch": "yarn build:ts:spec -- -w",
+    "build:ts:watch": "del ./dist/*.tsbuildinfo && tsc --p ./tsconfig.spec.json -w",
     "build:ts:spec": "tsc --b ./tsconfig.spec.json --force",
     "build:src": "npm-run-all build:sass:components build:ts",
-    "build:api": "wca analyze './dist/core/**/*.{js,ts}' --silent --format=json --outFile ./dist/core/custom-elements.json && wca analyze './dist/core/**/*.{js,ts}' --silent --format=vscode --outFile ./dist/core/vscode-html-custom-data.json",
+    "prebuild:api": "cd dist/core && yarn link",
+    "build:api": "wca analyze './dist/core/**/*.{js,ts}' --silent --format=json --outFile ./dist/core/custom-elements.json",
+    "postbuild:api": "cd dist/core && yarn unlink --no-save @clr/core",
```
Comment (coryrylan) [inline packages/core/package.json:28]: We link then unlink so the API tool can resolve the entrypoints since by default it does not respect the tsconfig path mapping.

File: packages/core/package.json [hunk 2]
```diff
@@ -13,19 +12,20 @@
...
-    "build": "npm-run-all -s build:tokens build:sass build:ts build:sass:min build:package build:clean build:api build:copy",
+    "build": "npm-run-all -s build:tokens build:sass build:ts build:sass:min build:package build:api build:copy",
     "build:watch": "npm-run-all -p build:sass:watch build:ts:watch",
-    "build:clean": "node ./package.js",
-    "build:package": "cpy '**/package.json' '../dist/core/' --cwd=./src --parents && cpy ./package.json ./dist/core/ && cpy ./README.md ./dist/core/; replace '@VERSION' $npm_package_version ./dist/core/package.json; replace '@VERSION' $npm_package_version ./dist/core/internal/utils/global.js",
+    "build:package": "cpy '**/package.json' '../dist/core/' --cwd=./src --parents && cpy ./package.json ./dist/core/ && cpy ./README.md ./dist/core/; replace '0.0.0' $npm_package_version ./dist/core/package.json; replace '@VERSION' $npm_package_version ./dist/core/internal/utils/global.js && node ./package.js",
     "build:ts": "tsc --b ./tsconfig.project.json --force",
-    "build:ts:watch": "yarn build:ts:spec -- -w",
+    "build:ts:watch": "del ./dist/*.tsbuildinfo && tsc --p ./tsconfig.spec.json -w",
```
Comment (coryrylan) [inline packages/core/package.json:23]: Run a project mode during watch as this speeds up rebuild times

File: packages/core/package.json [hunk 3]
```diff
@@ -42,40 +42,44 @@
...
+    "@storybook/addons": "6.0.0-beta.20",
+    "@storybook/cli": "6.0.0-beta.20",
+    "@storybook/web-components": "6.0.0-beta.20",
     "@types/jasmine": "3.3.13",
     "@types/jasmine-expect": "3.8.1",
     "@types/jasmine-matchers": "0.2.32",
     "@types/jasminewd2": "2.0.6",
     "@types/node": "^12.11.1",
     "@types/ramda": "0.26.1",
+    "@types/resize-observer-browser": "^0.1.3",
```
Comment (coryrylan) [inline packages/core/package.json:79]: This should be able to be removed in TS 4

File: packages/core/src/alert/alert-group.element.spec.ts [hunk 4]
```diff
@@ -78,7 +78,7 @@ describe('Alert groups – ', () => {
 
       layout = wrapper.getAttribute('cds-layout');
       expect(layout.includes('vertical')).toBe(true, 'Compact alert group should include vertical layout');
-      expect(layout.includes('no-wrap')).toBe(true, 'Compact alert group should include no-wrap layout');
+      expect(layout.includes('wrap:none')).toBe(true, 'Compact alert group should include wrap:none layout');
```
Comment (coryrylan) [inline packages/core/src/alert/alert-group.element.spec.ts:81]: Updated wrap to support params to support additional wrapping options (forms in next pr)

File: packages/core/src/button/base-button.element.scss [hunk 5]
```diff
@@ -105,8 +98,13 @@
...
   text-decoration: none !important;
   display: block;
-  height: 100%;
   color: inherit;
+  margin: calc(var(--padding) * -1);
+  padding: var(--padding);
+  min-width: var(--min-width);
+  line-height: 1em;
+  box-sizing: border-box !important;
+  outline: 0;
```
Comment (coryrylan) [inline packages/core/src/button/base-button.element.scss:107]: Fixes anchors to match the same size as the button so the touch target and focus styles are correct.

File: packages/core/src/icon-shapes/icon.element.ts [hunk 6]
```diff
@@ -179,7 +178,7 @@ export class CdsIcon extends IconMixinClass {
 
   protected render() {
     return html`
-      ${unsafeHTML(ClarityIcons.registry[this.shape])}
+      <span .innerHTML="${ClarityIcons.registry[this.shape]}"></span>
```
Comment (coryrylan) [inline packages/core/src/icon-shapes/icon.element.ts:181]: This fixes an issue with icons sometimes not rendering when re-rendered

File: packages/core/src/styles/layout/layout.stories.ts [hunk 7]
```diff
@@ -1201,7 +1399,7 @@ export const spacingMarginSides = () => {
 export const spacingMarginResponsive = () => {
   return html`
     <cds-demo spacing-margin cds-layout="vertical gap:xl">
-      <div><cds-placeholder cds-layout="m@sm:md">m@sm:md</cds-placeholder></div>
+      <div><cds-placeholder cds-layout="m@sm:md">m@sm</cds-placeholder></div>
```
Comment (coryrylan) [inline packages/core/src/styles/layout/layout.stories.ts:1402]: todo: fix missing syntax

File: packages/core/src/styles/layout/layout.stories.ts [hunk 8]
```diff
@@ -1159,7 +1357,7 @@ export const spacingPaddingSides = () => {
 export const spacingPaddingResponsive = () => {
   return html`
     <cds-demo spacing-padding cds-layout="vertical gap:xl">
-      <div cds-layout="p@sm:md"><cds-placeholder>p@sm:md</cds-placeholder></div>
+      <div cds-layout="p@sm"><cds-placeholder>p@sm</cds-placeholder></div>
```
Comment (coryrylan) [inline packages/core/src/styles/layout/layout.stories.ts:1360]: todo: fix missing syntax

File: packages/core/src/icon-shapes/icon.element.scss [hunk 9]
```diff
@@ -22,11 +22,12 @@
 }
 
 :host {
+  --color: #{$cds-token-color-neutral-700};
```
Comment (mathisscott) [inline packages/core/src/icon-shapes/icon.element.scss:25]: this was changed because icon colors were not being overwritten in complex components. please verify before merging. check the custom alert icons. it should be a light color.

File: packages/core/src/alert/alert-actions.element.scss [hunk 10]
```diff
@@ -2,6 +2,8 @@
 @import './../styles/mixins/utils';
 
 :host {
+  --action-hover-text-color: #{$cds-token-color-neutral-1000};
+  --action-text-color: #{$cds-token-color-neutral-700};
```
Comment (coryrylan) [inline packages/core/src/alert/alert-actions.element.scss:6]: Moved vars to initialize `:host` as they are required for the polyfill and the next version of the API generator

File: packages/core/src/alert/alert.element.ts [hunk 11]
```diff
@@ -280,7 +280,11 @@ export class CdsAlert extends AlertMixinClass {
         </span>
         <span
           class="alert-content-wrapper"
-          cds-layout="${getAlertContentLayout('wrapper', this.alertGroupType, this.parentGroupHasPager)}"
+          cds-layout="horizontal wrap:none ${getAlertContentLayout(
+            'wrapper',
+            this.alertGroupType,
+            this.parentGroupHasPager
+          )}"
```
Comment (coryrylan) [inline packages/core/src/alert/alert.element.ts:287]: The fixes for layouts were to prevent the alignments leaking into nested child layouts. So its more explicit and only applies to direct children of the layout `>`. Alerts was relying on the buggy behavior I accidentally introduced so this fixes it and required the layout to be more explicit. Tested all the alerts, alert groups and alert with placeholder pager.

File: packages/core/src/internal-components/close-button/close-button.element.scss [hunk 12]
```diff
@@ -29,6 +31,7 @@
   transform: translateY(#{$cds-token-space-size-1});
 }
 
-:host() cds-icon {
-  --color: var(--color);
+cds-icon {
+  --color: var(--icon-color);
+  cursor: pointer;
```
Comment (coryrylan) [inline packages/core/src/internal-components/close-button/close-button.element.scss:36]: For some reason the cursor is ignored unless explicitly on the icon in chrome. The specificity problem I found was due to this line `--color: var(--color);` as a var assignment will refer to its own scope instead of the one on the `:host`. The way to use the same name on the host and child is to have an intermediate var on `.private-host` that can pass it down.

File: packages/core/src/styles/layout/_container.scss [hunk 13]
```diff
@@ -4,7 +4,7 @@
 
 [cds-layout~='fill'],
 [cds-layout~='container:fill'] {
-  width: 100%;
+  width: 100% !important;
```
Comment (coryrylan) [inline packages/core/src/styles/layout/_container.scss:7]: Missing from pr split

File: packages/core/src/styles/layout/_type-horizontal.scss [hunk 14]
```diff
@@ -40,6 +40,10 @@
   margin: 0;
   min-height: 0;
 
+  &[cds-layout~='wrap:none'] {
+    flex-wrap: nowrap;
+  }
```
Comment (coryrylan) [inline packages/core/src/styles/layout/_type-horizontal.scss:45]: Missing from pr split

File: packages/core/src/styles/module.reset.scss [hunk 15]
```diff
@@ -6,4 +6,11 @@
 
 html {
   font-size: $cds-token-global-base-font-size;
+  box-sizing: border-box;
+}
+
+*,
+*:before,
+*:after {
+  box-sizing: inherit;
```
Comment (coryrylan) [inline packages/core/src/styles/module.reset.scss:15]: Missing from pr split

File: packages/core/tokens.yml [hunk 16]
```diff
@@ -246,7 +246,7 @@ props:
     type: 'px'
     category: 'typography'
   - name: body-color
-    value: hsl(198, 0%, 10%)
+    value: 'hsl(198, 0%, 10%)'
```
Comment (coryrylan) [inline packages/core/tokens.yml:249]: Missing quotes cause theo to generate colors in rgb randomly, not sure why.

File: packages/core/tokens-build.js [hunk 17]
```diff
@@ -84,7 +84,8 @@ theo.registerFormat('core-tokens-scss', result => {
         !prop.private || (prop.private && isRelativePxValue(prop))
           ? `${sassVariable}: ${cssProp};\n`
           : `${sassVariable}: ${value};\n`;
-      const staticVariable = `${sassVariable}-static: ${value};`;
+
+      const staticVariable = prop.category !== 'color' ? `${sassVariable}-static: ${value};` : '';
```
Comment (coryrylan) [inline packages/core/tokens-build.js:88]: Only generate static for measured values, colors should always stay dynamic for theming

File: packages/core/src/alert/alert.element.ts [hunk 18]
```diff
@@ -304,14 +308,15 @@ export class CdsAlert extends AlertMixinClass {
...
-            ></span>`
+          ? html`<slot
+              name="close-button"
+              cds-layout="${this.alertGroupType === 'banner' ? 'align:vertical-center' : 'p-t:xs'}"
+              ><cds-internal-close-button
+                icon-size="${this.alertGroupType === 'banner' ? '20' : '16'}"
+                @click="${() => this.closeAlert()}"
+                aria-label="${this.closeIconTitle}"
+              ></cds-internal-close-button
+            ></slot>`
```
Comment (coryrylan) [inline packages/core/src/alert/alert.element.ts:319]: Sorry going to revert this change, was leftover from testing

Comment (coryrylan) [reply packages/core/src/alert/alert.element.ts]: Reverted

File: packages/core/src/styles/layout/_type-horizontal.scss [hunk 19]
```diff
@@ -5,12 +5,17 @@
 @mixin generateGaps($breakpoint: null) {
   @each $size, $sizeValue in $cds-layout-sizes {
     [cds-layout~='horizontal'][cds-layout~='gap#{$breakpoint}:#{$size}'] {
-      margin: calc(-1 * #{$sizeValue}) 0 0 calc(-1 * #{$sizeValue});
       width: calc(100% + #{$sizeValue});
+      margin-bottom: calc(#{$sizeValue} * -1) !important;
+      margin-right: calc(#{$sizeValue} * -1);
```
[the remark this answers is not in the document]
Comment (coryrylan) [reply packages/core/src/styles/layout/_type-horizontal.scss:10]: I'll fix that, I think they both are supposed to
